tests/h/search/query_test.py

```diff
@@ -141,12 +141,87 @@ def search(self, search):
 
 
 class TestUriFilter(object):
+    @pytest.mark.parametrize("field", ("uri", "url"))
+    def test_filters_by_field(self, search, Annotation, field):
+        Annotation(target_uri="https://foo.com")
+        expected_ids = [Annotation(target_uri="https://bar.com").id]
+
+        result = search.run({field: "https://bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filters_on_whole_url(self, search, Annotation):
+        Annotation(target_uri="http://bar.com/foo")
+        expected_ids = [Annotation(target_uri="http://bar.com").id,
+                        Annotation(target_uri="http://bar.com/").id]
+
+        result = search.run({"url": "http://bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filter_matches_invalid_uri(self, search, Annotation):
+        Annotation(target_uri="https://bar.com")
+        expected_ids = [Annotation(target_uri="invalid-uri").id]
+
+        result = search.run({"uri": "invalid-uri"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filters_aliases_http_and_https(self, search, Annotation):
+        expected_ids = [Annotation(target_uri="http://bar.com").id,
+                        Annotation(target_uri="https://bar.com").id]
+
+        result = search.run({"url": "http://bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_returns_all_annotations_with_equivalent_uris(self, search, Annotation, storage):
+        # Mark all these uri's as equivalent uri's.
+        storage.expand_uri.side_effect = lambda _, x: [
```

Comment:
If you just want to set the return value of a callable you can use `.return_value = {value}` rather than needing to set `side_effect` to a lambda.